SE050F ECC keypair generation and shared secret fix

msalvini-egf opened the pull request:
Fix ECC part for SE050 in F configuration:
- revert commit db8ca286ea44786a19e344ca02b47cc8be04924e to have working keypair generation
- fix shared secret generation in another way, passing the ECDH algo (the only one used) to ecc_get_key_size() 
<!--
    If you are new to submitting pull requests to OP-TEE, then please have a
    look at the list below and tick them off before submitting the pull request.

    1. Read our contribution guidelines:
         https://optee.readthedocs.io/en/latest/general/contribute.html

    2. Read the contribution section in Notice.md and pay extra attention to the
       "Developer Certificate of Origin" in the contribution guidelines.

    3. You should run checkpatch preferably before submitting the pull request.

    4. When everything has been reviewed, you will need to squash, rebase and
       add tags like `Reviewed-by`, `Acked-by`, `Tested-by` etc. More details
       about this can also be found on the link provided above.

    NOTE: This comment will not be shown in the pull request, so no harm keeping
    it, but feel free to remove it if you like.
-->

ldts commented:
> Tested RNG functionality. LGTM

thanks @mike-scott , I removed that commit from this PR now

msalvini-egf committed:
Revert "se050: ecc: SE050-F shared secret"

This reverts commit db8ca286ea44786a19e344ca02b47cc8be04924e.
Revert is done because this commit fixed shared secret generation
but broke the key generation for SE050-F configuration.
Let's try to fix both in another way with next commit.

Signed-off-by: Mauro Salvini <mauro.salvini@elettronicagf.it>

msalvini-egf committed:
se050: ecc: disable SE050-F shared secret

The SE050-F does not support shared secret generation.
Fix the handling of this operation to fallback to its software
implementation, passing the algo parameter with fixed value (ECDH)
to ecc_get_key_size() instead of 0 (that means ignore parameter).
We can do this because the used secret generation algo is always ECDH
(see fourth parameter of sss_se05x_derive_key_context_init() call done some
code lines after the patched line).

Signed-off-by: Mauro Salvini <mauro.salvini@elettronicagf.it>
